crates/core/protobuf/node_ctl_svc.proto

```diff
@@ -27,6 +29,21 @@ service NodeCtlSvc {
 
   // Returns the cluster health from the point of view of this node.
   rpc ClusterHealth(google.protobuf.Empty) returns (ClusterHealthResponse);
+
+  // Metadata proxy functions
+
+  // Get a versioned kv-pair
+  rpc MetadataGet(MetadataGetRequest) returns (MetadataGetResponse);
+
+  // Get the current version for a kv-pair
+  rpc MetadataGetVersion(MetadataGetRequest)
+      returns (MetadataGetVersionResponse);
+
+  // Puts the given kv-pair into the metadata store
+  rpc MetadataPut(MetadataPutRequest) returns (google.protobuf.Empty);
+
+  // Deletes the given kv-pair
+  rpc MetadataDelete(MetadataDeleteRequest) returns (google.protobuf.Empty);
```

Comment:
Wondering whether splitting those methods into a `MetadataProxySvc` would help distinguishing between `GetMetadata` and `MetadataGet`.

Reply:
Yeah definitely. I was considering deprecating out `MetadataGet` at first but realized it has completely different semantic. Definitely the freedom to move this to a different service is much better solution.

Reply:
@tillrohrmann I have split the metadata proxy into its own service as advised. Feel free to take another look if you like :)